Assets/PatchKit Patcher/Scripts/UI/DownloadSpeed.cs

```diff
@@ -50,39 +64,61 @@ private string GetDownloadSpeedFormated(OverallStatus status)
             }
         }
 
-        private string FormatDownloadSpeedMegabytes(double kbPerSecond)
+        private static string FormatDownloadSpeedMegabytes(double kbPerSecond)
         {
             return FormatDownloadSpeed(kbPerSecond / 1024.0) + " MB/sec.";
         }
 
-        private string FormatDownloadSpeedKilobytes(double kbPerSecond)
+        private static string FormatDownloadSpeedKilobytes(double kbPerSecond)
         {
             return FormatDownloadSpeed(kbPerSecond) + " KB/sec.";
         }
 
-        private string FormatDownloadSpeed(double s)
+        private static string FormatDownloadSpeed(double s)
         {
             return s.ToString("#,#0.0");
         }
 
-        private float GetRemainingTime(OverallStatus status)
+        private static string GetFormattedRemainingTime(double? remainingTime)
         {
-            float remainingBytes = status.DownloadTotalBytes - status.DownloadBytes;
-            return remainingBytes / (float) status.DownloadBytesPerSecond;
+            if (!remainingTime.HasValue)
+            {
+                return string.Empty;
+            }
+
+            var span = TimeSpan.FromSeconds(remainingTime.Value);
+            return span.TotalDays > 1.0
+                ? string.Format("{0:0} day", span.TotalDays) + GetPlural(span.TotalDays)
+                : span.TotalHours > 1.0
+                    ? string.Format("{0:0} hour", span.TotalHours) + GetPlural(span.TotalHours)
+                    : span.TotalMinutes > 1.0
+                        ? string.Format("{0:0} minute", span.TotalMinutes) + GetPlural(span.TotalMinutes)
+                        : span.TotalSeconds > 1.0
+                            ? string.Format("{0:0} second", span.TotalSeconds) +
+                              GetPlural(span.TotalSeconds)
+                            : "a moment";
         }
 
-        private string GetRemainingTimeFormated(float value)
+        private static double? GetRemainingTime(long bytes, long totalBytes, double bytesPerSecond)
         {
-            TimeSpan span = TimeSpan.FromSeconds(value);
-            return span.Days > 0 ? string.Format("{0:0} day", span.Days) + GetPlural(span.Days) :
-                span.Hours > 0 ? string.Format("{0:0} hour", span.Hours) + GetPlural(span.Hours) :
-                span.Minutes > 0 ? string.Format("{0:0} minute", span.Minutes) + GetPlural(span.Minutes) :
-                span.Seconds > 0 ? string.Format("{0:0} second", span.Seconds) + GetPlural(span.Seconds) : "a moment";
+            if (bytesPerSecond <= 0.0)
+            {
+                return null;
+            }
+
+            double remainingBytes = totalBytes - bytes;
+
+            if (remainingBytes <= 0)
+            {
+                return null;
+            }
+
+            return remainingBytes / bytesPerSecond;
```

Comment:
Please change it to `Math.Max(remainingBytes / bytesPerSecond, 0)`. We don't want to see negative values here.

Reply:
There won't be negative numbers here - `if (bytesPerSecond <= 0.0)` and `if (remainingBytes <= 0)` are protecting from such situation.